festim/hydrogen_transport_problem.py

```diff
@@ -32,11 +32,16 @@ def __init__(
         self.solver_parameters = solver_parameters
         self.exports = exports
 
+        self.dx = None
+        self.ds = None
+        self.function_spaces = []
+
     def initialise(self):
         """Initialise the model. Creates suitable function
         spaces, the subdomains...
         """
 
-        # geometry
-        self.geometry.define_function_space()
-        self.geometry.define_subdomains()
+        self.function_spaces.append(self.geometry.define_function_space())
```

Comment:
See my other comments, `define_function_space` shouldn't be a method of geometry but a method of `HydrogenTransportProblem`

```suggestion
        self.function_space = self.define_function_space()
```